managers/s3.py

```diff
@@ -37,11 +42,49 @@ def createS3Bucket(self, bucketName, bucketPermissions):
         raise S3Error('Unable to create bucket in s3')
 
 
-    def createManifestInS3(self, manifestPath, manifestJSON, s3_bucket: str):
+    def store_pdf_manifest(self, record, bucket_name, flags=None, requires_login: bool=False):
+        record_id = record.source_id.split('|')[0]
+        pdf_part = next(filter(lambda part: part.file_type == 'application/pdf', record.get_parts()), None)
+
+        if pdf_part is not None:
+            if requires_login:
+                manifest_path = f'manifests/publisher_backlist/{record.source}/{record_id}.json'
+            else:
+                manifest_path = f'manifests/{record.source}/{record_id}'
+            manifest_url = get_stored_file_url(storage_name=bucket_name, file_path=manifest_path)
+            manifest_json = self.generate_manifest(record=record, source_url=pdf_part.url, manifest_url=manifest_url)
+
+            self.create_manifest_in_s3(manifest_path=manifest_path, manifest_json=manifest_json, s3_bucket=bucket_name)
+
+            if flags:
+                record.has_part.insert(0, Part(
+                    index=pdf_part.index,
+                    url=manifest_url,
+                    source=record.source,
+                    file_type='application/webpub+json',
+                    flags=flags
+                ).to_string())
```

Comment:
I think my question for CLACSO is if the PDF is not readable, do we need to store the manifest until it is readable? In other words, do we need to call this function altogether? 

This would mean that flags should not be None and we can remove this if.

Reply:
See my comment below as well - we could perhaps just do FileFlags(reader=True) when adding the manifest has_part.

Reply:
We don't need to call the store_pdf_manifest for CLACSO in its current state since we don't want these books to be readable currently. We already have CLACSO manifests in the production bucket currently but I can remove the store_pdf_manifest function from CLACSO for now and I'll remove the flag conditional. What are your thoughts?